doc/user/content/sql/create-source/kafka.md

```diff
@@ -202,50 +202,42 @@ Note that:
 
 #### Headers
 
-Message headers can be exposed via the `INCLUDE HEADERS` option. They are included
-as a column (named `headers` by default) containing a [`list`](/sql/types/list/)
-of records of type `(key text, value bytea)`.
+Message headers can be exposed via the `INCLUDE HEADER key AS name` option.
+The `bytea` value of the header is automatically parsed into an UTF-8 string. To expose the raw `bytea` instead, the `BYTES` option can be used.
 
-The following example demonstrates use of the `INCLUDE HEADERS` option.
+The following example demonstrates use of the `INCLUDE HEADER` option.
 
 ```sql
 CREATE SOURCE kafka_metadata
   FROM KAFKA CONNECTION kafka_connection (TOPIC 'data')
   FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY CONNECTION csr_connection
-  INCLUDE HEADERS
+  INCLUDE HEADER 'c_id' AS client_id, HEADER 'key' AS encryption_key BYTES,
   ENVELOPE NONE
   WITH (SIZE = '3xsmall');
 ```
 
-To retrieve the value of an individual header in a message, you can use standard
-SQL techniques for working with [`list`](/sql/types/list) and
-[`bytea`](/sql/types/bytea) types. The following example parses the UTF-8
-encoded `client_id` header of the messages from the Kafka topic. Messages
-without a `client_id` header result in null values (`"\N"`) for the parsed
-attribute.
+The headers can then be queried as any other column of the message.
```

Comment:
> Otherwise, I'd rather not drop how to unnest the result of the INCLUDE HEADERS option.

The `unnest` has terrible memory requirements. It's okay to use `INCLUDE HEADERS` if you just want a bag of arbitrary header keys/values, but if you're looking to extract any _specific_ header keys, `INCLUDE HEADER` is the way to go.

If we do want to keep an `INCLUDE HEADERS` example, I wouldn't recommend using `unnest` in that example—instead just a simple example showing a `headers` column with a map of 2-3 headers in it.

Reply:
I think we should remove it entirely. We should not show `unnest` and I'm not really sure what value the example adds if it does not include that.